### PR TITLE
[Python BQ] Substitute final destination schema when no input schema is specified

### PR DESCRIPTION
**Pull request by ahmedabu98**
When users perform BQ file load writes, we depend on them to provide an input schema. This is a limitation because for larger loads we create temp tables. Although we don't need a schema to write to an existing table, we do need it when creating a new table. Without an input schema, this use case fails.

To fix this, we should fetch the final destination's schema when necessary.

### CODE REVIEW

**Comment by ahmedabu98** [comment]
R: @shunping

File: sdks/python/apache_beam/io/gcp/bigquery_file_loads_test.py [hunk 1]
```diff
@@ -903,6 +903,68 @@ def setUp(self):
...
+
+    args = self.test_pipeline.get_full_options_as_args(
+        on_success_matcher=all_of(verifiers))
+
+    with beam.Pipeline(argv=args) as p:
+      # 0...4 going to table 1
+      # 5...9 going to table 2
+      items = [{"col_1": i} for i in range(5)]
+      items.extend([{"col_2": i} for i in range(5, 10)])
+      _ = (
```
Comment (shunping) [inline sdks/python/apache_beam/io/gcp/bigquery_file_loads_test.py:960]: Could we add a comment here to explicitly point out something like "calling WriteToBigQuery without specifying the schema parameter"?

**Comment by shunping** [comment]
Thanks! LGTM with some suggestion about adding comments in the code

File: sdks/python/apache_beam/io/gcp/bigquery_file_loads_test.py (hunk 1 above)
Comment (ahmedabu98) [reply sdks/python/apache_beam/io/gcp/bigquery_file_loads_test.py:960]: That's included in the test name `test_batch_copy_jobs_with_no_input_schema`.

Comment (shunping) [reply sdks/python/apache_beam/io/gcp/bigquery_file_loads_test.py:960]: ok, sgtm

**Pull request merged**